packages/material-ui/src/Grid/Grid.js

```diff
@@ -65,20 +66,63 @@ function generateGrid(globalStyles, theme, breakpoint) {
   }
 }
 
+// Same logic as /packages/material-ui-system/src/spacing.js
+function getTransformer(theme) {
+  const themeTransformer =
+    theme.spacing && theme.spacing.unit != null ? theme.spacing.unit : theme.spacing || 8;
+
+  if (typeof themeTransformer === 'number') {
+    return abs => themeTransformer * abs;
+  }
+
+  if (Array.isArray(themeTransformer)) {
```

Comment:
Why do we need an extra overload for array? Isn't this easily implementable in userland:

```diff
- theme.spacing = [1, 2, 4, 8, 16]
+ theme.spacing = index => [1, 2, 4, 8, 16][index]
```
I see the use case for a constant spacing and a customized spacing. I don't understand why we are opinionated about the data structure for customized spacing and add another way of customizing the spacing. 

This is also a general issue I have with `theme.spacing`. Currently these are equivalent in `Grid`
```js
theme.spacing = 5
theme.spacing = { unit: 5 }
```
or 
```js
theme.spacing = abs => 2 ** abs
theme.spacing.unit = abs => 2 ** abs
```
Can we unify the `theme.spacing` API before we allow all these overloads?